formats/protobuf/commonTest/src/kotlinx/serialization/protobuf/PackedArraySerializerTest.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.protobuf
+
+import kotlinx.serialization.*
+import kotlin.test.*
+
+class PackedArraySerializerTest {
+
+    abstract class BaseFloatArrayCarrier {
+        abstract val createdAt: ULong
+        abstract val vector: FloatArray
+
+        override fun equals(other: Any?): Boolean {
+            if (this === other) return true
+            if (other !is BaseFloatArrayCarrier) return false
+
+            if (createdAt != other.createdAt) return false
+            if (!vector.contentEquals(other.vector)) return false
+
+            return true
+        }
+
+        override fun hashCode(): Int {
+            var result = createdAt.hashCode()
+            result = 31 * result + vector.contentHashCode()
+            return result
+        }
+    }
+
+    @Serializable
+    class PackedFloatArrayCarrier(
+        @ProtoNumber(2)
+        override val createdAt: ULong,
+        @ProtoPacked
+        @ProtoNumber(3) override val vector: FloatArray
+    ) : BaseFloatArrayCarrier()
+
+    @Serializable
+    class NonPackedFloatArrayCarrier(
+        @ProtoNumber(2)
+        override val createdAt: ULong,
+        @ProtoNumber(3) override val vector: FloatArray
+    ) : BaseFloatArrayCarrier()
+
+    /**
+     * Test that when packing is specified the array is encoded as packed
+     */
+    @Test
+    fun testEncodePackedFloatArrayProtobuf() {
+        val obj = PackedFloatArrayCarrier(1234567890L.toULong(), floatArrayOf(1f, 2f, 3f))
+        val s = ProtoBuf.encodeToHexString(PackedFloatArrayCarrier.serializer(), obj).uppercase()
+        assertEquals("""10D285D8CC041A0C0000803F0000004000004040""", s)
+    }
+
+    /**
+     * Test that when packing is not specified the array is not encoded as packed. Note that protobuf 3
+     * should encode as packed by default. The format doesn't allow specifying versions at this point so
+     * the default remains the original.
+     */
+    @Test
+    fun testEncodeNonPackedFloatArrayProtobuf() {
+        val obj = PackedFloatArrayCarrier(1234567890L.toULong(), floatArrayOf(1f, 2f, 3f))
+        val s = ProtoBuf.encodeToHexString(PackedFloatArrayCarrier.serializer(), obj).uppercase()
+        assertEquals("""10D285D8CC041A0C0000803F0000004000004040""", s)
```

Comment:
I may be missing something, but why this string and string in `testEncodePackedFloatArrayProtobuf` are exactly the same?

Reply:
Good catch. Actually there was an error in the previous packed test (using the wrong class) where this was copied from.